File: 2ano/1semestre/aed/aula01/ProgA.c
#include <stdio.h>

void print_array(char s[], int arr[], int arr_size) {
    printf("%s:\n", s);
    for (int i=0; i<arr_size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

void cumSum(int a[], int b[], int arr_size) {
    int c = 0;
    for (int i=0; i<arr_size; i++) {
        c += a[i];
        b[i] = c;
    }
}

int main() {
    int a[12] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    print_array("a", a, sizeof(a)/sizeof(*a));
    int b[12] = {0};
    cumSum(a, b, sizeof(a)/sizeof(*a));
    print_array("b", b, sizeof(b)/sizeof(*b));
}

File: 2ano/1semestre/aed/aula01/ProgA.java

/*
Crie um programa em C equivalente a este em Java.
*/

public class ProgA {
    
    public static void main(String[] args) {
        int[] a = {31,28,31,30,31,30,31,31,30,31,30,31};
        printArray("a", a);

        int[] b = new int[12];
        cumSum(a, b);
        printArray("b", b);
    }

    public static void printArray(String s, int[] a) {
        System.out.println(s + ":");
        for (int i=0; i<a.length; i++) {
            System.out.print(a[i]+" ");
        }
        System.out.println();
    }

    public static void cumSum(int[] a, int[] b) {
        int c = 0;
        for (int i=0; i<a.length; i++) {
            c += a[i];
            b[i] = c;
        }
    }
}

